vsg/vhdlFile/vhdlFile.py

```diff
@@ -46,19 +46,36 @@ class vhdlFile():
 
        fileobject
     '''
-    def __init__(self, filecontent, sFilename=None):
-        self.filecontent = filecontent
+    def __init__(self, sFilename=None):
+        self.filename = sFilename
         self.hasArchitecture = False
         self.hasEntity = False
         self.lAllObjects = []
-        self.filename = sFilename
+        if self.filename:
+            self.filecontent = self._read_vhdlfile()
+        else:
+            self.filecontent = ['']
         self.dIndentMap = None
         self.lOpenPragmas = ['--vhdl_comp_off']
         self.lClosePragmas = ['--vhdl_comp_on']
         self.dVars = {}
         self.dVars['pragma'] = False
         self._processFile()
 
+    def _read_vhdlfile(self):
```

Comment:
I had originally included a function like `_read_vhdlfile` in this class.  Then I wondered if VSG could be included in other python tools, so I removed it and switch to a list of strings.

The thinking was VSG could be imported, and in the user's app they would create whatever VHDL they wanted.  Then they could pass it to VSG and get a formatted version back.

Can we move this to a `utils.py` file instead?

Reply:
Makes sense, I'll move it to `vhdlFile/utils.py` then.